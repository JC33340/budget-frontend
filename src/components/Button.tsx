type ButtonType = {
    text: string;
    handleClick: () => void;
    className?: string;
};

const Button = ({ text, handleClick, className }: ButtonType) => {
    return (
        <button
            data-testid="button-testid"
            className={`p-2 border-2 border-blue rounded-md bg-blue text-white hover:bg-white hover:text-blue transition-all ease-in-out ${className}`}
            onClick={handleClick}
        >
            {text}
        </button>
    );
};

export default Button;
