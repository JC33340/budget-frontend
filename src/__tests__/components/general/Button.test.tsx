import Button from '../../../components/Button';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

test('snapshot test', () => {
    render(<Button text="test" handleClick={() => {}} />);
    const button = screen.getByTestId('button-testid');
    expect(button).toMatchSnapshot();
});

test('testing button functionality', async () => {
    const user = userEvent.setup();
    const testFunc = jest.fn();
    render(<Button text="test-text" handleClick={testFunc} />);
    const button = screen.getByTestId('button-testid');
    await user.click(button);
    expect(testFunc).toHaveBeenCalledTimes(1);
});
