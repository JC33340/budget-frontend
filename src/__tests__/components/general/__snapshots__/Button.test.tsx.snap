// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<button
  class="p-2 border-2 border-blue rounded-md bg-blue text-white hover:bg-white hover:text-blue transition-all ease-in-out undefined"
  data-testid="button-testid"
>
  test
</button>
`;
